rp2040-hal/src/rosc.rs

```diff
@@ -97,20 +92,21 @@ impl RingOscillator<Enabled> {
         self.device.randombit.read().randombit().bit()
     }
 
-    /// Put the ROSC in DORMANT state.
+    /// Put the ROSC in DORMANT state. The method returns after the processor awakens.
+    ///
+    /// After waking up from the DORMANT state, ROSC restarts in approximately 1µs.
     ///
     /// # Safety
     /// This method is marked unsafe because prior to switch the ROSC into DORMANT state,
     /// PLLs must be stopped and IRQs have to be properly configured.
     /// This method does not do any of that, it merely switches the ROSC to DORMANT state.
+    /// It should only be called if this oscillator is the clock source for the system clock.
     /// See Chapter 2, Section 16, §5) for details.
-    pub unsafe fn dormant(self) -> RingOscillator<Dormant> {
+    pub unsafe fn dormant(self) {
```

Comment:
This consumes the `RingOscillator<_>` without returning anything.
Is this intended?

Reply:
Oops, well spotted. I didn't have my rust brain fully switched on when removing the `Dormant` as you suggested earlier @ithinuel. :-} I'll change it to borrow as intended.